src/controllers/AbstractController.ts

```diff
@@ -48,31 +53,76 @@ export abstract class AbstractController {
             if(err) {
                 return res.status(400).json({error: err.message})
             } else {
+                if(this.wss) {
+                    this.broadcast({action: "CREATE", value: artifact})
+                }
                 return res.status(201).json(artifact)
             }
         })
     }
 
     public delete = async(req: any, res: any) => {
-        await  this.model.deleteOne({_id: req.params["_id"]}, (err) => {
+        await  this.model.findOneAndDelete({_id: req.params["_id"]}, {returnOriginal: true, useFindAndModify: false}, (err, doc) => {
             if(err) {
                 return res.status(400).json({error: err.message})
-            } else {
-                return res.status(202).send()
+            }
+            if(!doc) {
+                return res.status(404).json({error: "not found"})
+            }
+            else {
+                if(this.wss) {
+                    this.broadcast({action: "DELETE", value: req.params["_id"]})
```

Comment:
Are you sure it's enough to just send the id? Shouldn't we send also the entity that had been deleted? How will the client know what had been deleted?

Reply:
The client suppose to save the documents by ID.
For now we don't have an implementation for the client, this may be changed later.